pdr_backend/pdr_dashboard/util/db.py

```diff
@@ -63,12 +65,70 @@ def start_date_ms(self) -> int:
         return UnixTimeMs.from_dt(self.start_date) if self.start_date else None
 
     def set_start_date_from_period(self, period: int):
-        start_dt = datetime.now() - timedelta(days=period) if int(period) > 0 else None
+        start_dt = (
+            datetime.now(tz=timezone.utc) - timedelta(days=period)
+            if int(period) > 0
+            else None
+        )
         self.start_date = start_dt
 
+    @enforce_types
+    def _check_cache_query_data(
```

Comment:
btw I think a good usage of the cache would also be with formatted dataframes. We do formatting a lot, and it implies copying and formatting a dataframe under certain rules. Maybe we could also define the values for the formatted version of the result and add them to the cache as well